datafusion/physical-expr/src/aggregate/average.rs

```diff
@@ -488,12 +488,10 @@ where
                 .map(|(sum, count)| (self.avg_fn)(sum, count))
                 .collect::<Result<Vec<_>>>()?;
             PrimitiveArray::new(averages.into(), Some(nulls)) // no copy
+                .with_data_type(self.return_data_type.clone())
         };
 
-        // fix up decimal precision and scale for decimals
-        let array = adjust_output_array(&self.return_data_type, Arc::new(array))?;
```

Comment:
I guess this was add before `with_data_type` is introduced as API.

Reply:
Oh yea, just realized this was added recently.